### PR TITLE
[mqtt] Add additional transport for local MQTT communication

### PR DESCRIPTION
**Pull request by dmolokanov**
Add additional transport for local MQTT communication on `localhost:1882`.
Run additional transport only when `edgehub` feature activated.
Add authenticator that allows any connection on localhost.

### CODE REVIEW

File: mqtt/mqtt-broker-core/src/auth/authentication.rs [hunk 1]
```diff
@@ -43,34 +34,89 @@ pub trait Authenticator {
...
+        Self {
+            client_id,
+            peer_addr,
+            username: None,
+            password: None,
+            certificate: None,
+        }
+    }
+
+    pub fn with_username(&mut self, username: impl Into<String>) -> &mut Self {
```
Comment (vadim-kovalyov) [inline mqtt/mqtt-broker-core/src/auth/authentication.rs:62]: q: nit: so for builder pattern should we use "with_xxxx" everywhere?

File: mqtt/mqtt-broker-core/src/auth/authentication.rs [hunk 2]
```diff
@@ -82,44 +128,50 @@ pub struct DefaultAuthenticator;
...
 
 #[cfg(test)]
 mod tests {
+    use std::net::SocketAddr;
+
     use matches::assert_matches;
 
-    use crate::auth::{
-        authenticate_fn_ok, AuthId, Authenticator, Credentials, DefaultAuthenticator,
+    use super::{
```
Comment (vadim-kovalyov) [inline mqtt/mqtt-broker-core/src/auth/authentication.rs:142]: I thought we were not in favor of `super` :)

File: mqtt/mqtt-edgehub/src/auth/edgehub.rs [hunk 3]
```diff
@@ -57,35 +61,31 @@ impl Authenticator for EdgeHubAuthenticator {
...
 #[derive(Serialize)]
 #[serde(rename_all = "snake_case")]
-pub struct EdgeHubAuthRequest {
-    version: String,
-    username: Option<String>,
-    password: Option<String>,
+pub struct EdgeHubAuthRequest<'a> {
+    version: &'a str,
+    username: Option<&'a str>,
+    password: Option<&'a str>,
```
Comment (vadim-kovalyov) [inline mqtt/mqtt-edgehub/src/auth/edgehub.rs:76]: q: you didn't want to clone values?

File: mqtt/mqtt-edgehub/src/auth/local.rs [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+pub struct LocalAuthenticator;
+
+impl LocalAuthenticator {
+    pub fn new() -> Self {
+        Self::default()
+    }
+}
+
+#[async_trait]
+impl Authenticator for LocalAuthenticator {
```
Comment (vadim-kovalyov) [inline mqtt/mqtt-edgehub/src/auth/local.rs:17]: can you please add some comments on why we need this LocalAuthenticator?

File: mqtt/mqtt-broker-core/src/auth/authentication.rs (hunk 1 above)
Comment (dmolokanov) [reply mqtt/mqtt-broker-core/src/auth/authentication.rs:62]: this specific case is different: it an object that optional fields set in "builder-like" style. usually the builder doesn't expose getters.

But, seems like a good idea to have similar semantics across all the code

File: mqtt/mqtt-edgehub/src/auth/edgehub.rs (hunk 3 above)
Comment (dmolokanov) [reply mqtt/mqtt-edgehub/src/auth/edgehub.rs:76]: correct

File: mqtt/mqtt-edgehub/src/auth/local.rs (hunk 4 above)
Comment (dmolokanov) [reply mqtt/mqtt-edgehub/src/auth/local.rs:17]: There is a comment when we actually use this authenticator.

File: mqtt/mqtt-broker-core/src/auth/authentication.rs (hunk 2 above)
Comment (dmolokanov) [reply mqtt/mqtt-broker-core/src/auth/authentication.rs:142]: Yeah, I am moving code across modules for the last few weeks, and changed my opinion:
1. when in unit tests we use **parent** module items: struct, enums, functions - use `super`
2. whatever outside of parent module, use `crate::module::Struct`

**Pull request merged**